sdk/core/Azure.Core/src/Pipeline/Internal/DefaultRequestFailedDetailsParser.cs

```diff
@@ -21,23 +22,15 @@ public static bool TryParseDetails(Response response, out ResponseError? error,
 
             try
             {
-                // The response content is buffered at this point.
-                string? content = response.Content.ToString();
-
                 // Optimistic check for JSON object we expect
-                if (content == null || !content.StartsWith("{", StringComparison.OrdinalIgnoreCase))
+                if (response.Content.Length <= 0 || response.Content.ToMemory().Span[0] != (byte)'{')
```

Comment:
Clever.  I forgot that `ToMemory` doesn't allocate and just passes through the underying buffer.